### PR TITLE
Merge branch "enis" into "dev"

### PR DESCRIPTION
**Pull request by richardpen**
### Summary

Merge the enis branch into dev branch

### Testing

- [x] Builds on Linux (`make release`)
- [x] Builds on Windows (`go build -out amazon-ecs-agent.exe ./agent`)
- [x] Unit tests on Linux (`make test`) pass
- [x] Unit tests on Windows (`go test -timeout=25s ./agent/...`) pass
- [x] Integration tests on Linux (`make run-integ-tests`) pass
- [ ] Integration tests on Windows (`.\scripts\run-integ-tests.ps1`) pass
- [ ] Functional tests on Linux (`make run-functional-tests`) pass
- [ ] Functional tests on Windows (`.\scripts\run-functional-tests.ps1`) pass

New tests cover the changes: 

### Licensing

This contribution is under the terms of the Apache 2.0 License: 
yes

### CODE REVIEW

File: agent/api/eni.go [hunk 1]
```diff
@@ -0,0 +1,145 @@
+// Copyright 2014-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (vsiddharth) [inline agent/api/eni.go]: Shouldn't this be only 2017

File: agent/api/enistatus.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
+// Copyright 2014-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (vsiddharth) [inline agent/api/enistatus.go]: same comment as above

File: agent/api/json.go [hunk 3]
```diff
@@ -1,224 +0,0 @@
-// Copyright 2014-2015 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (vsiddharth) [inline agent/api/json.go]: Why is the file removed

File: agent/ecscni/plugin_test.go [hunk 4]
```diff
@@ -0,0 +1,114 @@
+package ecscni
```
Comment (vsiddharth) [inline agent/ecscni/plugin_test.go]: Missing copyright

File: agent/engine/task_manager_test.go [hunk 5]
```diff
@@ -44,6 +45,9 @@ func TestContainerNextState(t *testing.T) {
 		// NONE -> RUNNING transition is allowed and actionable, when desired is Running
 		// The exptected next status is Pulled
 		{api.ContainerStatusNone, api.ContainerRunning, api.ContainerPulled, true, true},
+		// NONE -> RESOURCES_PROVISIONED transition is allowed and actionable, when desired
+		// is Running. The exptected next status is Pulled
```
Comment (vsiddharth) [inline agent/engine/task_manager_test.go]: nit: typo

File: agent/api/json.go (hunk 3 above)
Comment (richardpen) [inline agent/api/json.go]: See this [commit](https://github.com/aws/amazon-ecs-agent/commit/1af592561916daae8b0b51a7bda99bea73533e1a)

**Comment by richardpen** [comment]
closing this one, will have another one addressing this.

**Pull request closed without merging**